core/src/components/button/button.ionic.scss

```
@@ -53,45 +53,61 @@
   --ripple-color: var(--background-activated);
 }
 
+:host(.button-solid.ion-color) {
+  --background-activated: #{globals.current-color(shade)};
+  --background-hover: #{globals.current-color(shade)};
+  --background: #{globals.current-color(base)};
+  --color: #{globals.current-color(contrast)};
+}
+
 // Outline Button
 // --------------------------------------------------
 
 :host(.button-outline) {
   --border-width: #{globals.$ion-border-size-025};
   --border-style: #{globals.$ion-border-style-solid};
-  --background-activated: #{globals.$ion-bg-neutral-subtlest-press};
+  --background-activated: #{globals.$ion-bg-primary-subtle-press};
```

We can use the `ion-color` function to [access the subtle colors](https://github.com/ionic-team/ionic-framework/blob/9399d46e2d813a2e247138e2c8e04a0416bd6a7d/core/src/themes/functions.color.scss#L32).

```suggestion
  --background-activated: #{globals.ion-color(primary, shade, $subtle: true)};
```